packages/utils/src/activity.js

```diff
@@ -44,12 +32,30 @@ export function getAncestors(activities, activity) {
   return [...ancestors, parent];
 }
 
-export function toTreeFormat(activities, targetLevels, parentId = null, level = 1) {
-  return getOutlineChildren(activities, parentId).map(activity => ({
+export function getOutlineChildren(activities, parentId, schema) {
```

Comment:
schema config module might be a better place (signature might be simpler, no need to pass schema)